### ISSUE TITLE
Stub matches fail if stub created with URL params in the supplied URL

### ISSUE THREAD

**Issue by tmertens** [oesmith/puffing-billy#42]
If you create a stub using a URL with query parameters, then the stub on the exact same URL/query string fails:
```ruby
proxy.stub('http://example.test/index?some=params')
proxy.find_stub('GET','http://example.test/index?some=params') #=> returns nil, no stub found
```

The reason this occurs is because when a stub is created, it stores the full URL passed in to the initializer as `@url`:

https://github.com/oesmith/puffing-billy/blob/master/lib/billy/proxy_request_stub.rb#L5-L10
```ruby
    def initialize(url, options = {})
      @options = {:method => :get}.merge(options)
      @method = @options[:method].to_s.upcase
      @url = url
      @response = {code: 204, headers: {}, text: ""}
    end
```

However, when `ProxyRequestStub#matches?` substrings the passed in URL to strip off the query parameters and then tries to match the first part of the resulting URL sans parameters with the stub:

https://github.com/oesmith/puffing-billy/blob/master/lib/billy/proxy_request_stub.rb#L55-L63
```ruby
    def matches?(method, url)
      if method == @method
        if @url.is_a?(Regexp)
          url.match(@url)
        else
          url.split('?')[0] == @url
        end
      end
    end
```

If the original intent of this is correct, then the stub should probably strip off the query parameters when it stores the URL in the initializer.

**Comment by gkopylov** [comment]
@tmertens I added option for this in Billy configuration and create this PR https://github.com/oesmith/puffing-billy/pull/93

**Comment by ronwsmith** [comment]
Addressed in v0.5.0
